hnn_core/network.py

```diff
@@ -363,7 +378,8 @@ def __init__(self, params, add_drives_from_params=False,
                 stacklevel=1)
 
         # Source dict of names, first real ones only!
-        cell_types = {
+        # adding self before cell_types to make it an instance attribute
+        self.cell_types = {
```

Comment:
We may want to consider turning this into an `OrderedDict` to be explicit about order mattering here. Either that or we need to find a way to make order at this point irrelevant.

Reply:
Resolved in conversation with @wagdy88: our solution will be to maintain the original order of `Network.gid_ranges` after renaming, but not necessarily `Network.cell_types` or `Network.pos_dict`.